File: CPP/3273_두수의합.cpp
#include <iostream>
#include <algorithm>
using namespace std;

int main() {
    int n, arr[100000], sum_n;
    cin >> n;
    for (int i = 0; i < n; ++i) cin >> arr[i];
    cin >> sum_n;
    sort(arr, arr + n);
    int hap, left = 0, right = n - 1, cnt = 0;
    while (left != right) {
        hap = arr[left] + arr[right];
        if (sum_n == hap) {
            cnt += 1;
            left += 1;
        } else if (sum_n > hap) {
            left += 1;
        } else if (sum_n < hap) {
            right -= 1;
        }
    }
    cout << cnt << '\n';
    return 0;
}

File: Python/3273_두수의합.py
N = int(input())
arr = sorted(list(map(int, input().split())))
result = int(input())
cnt, left, right = 0, 0, N - 1
while left != right:
    sum_N = arr[left] + arr[right]
    if sum_N == result:
        cnt += 1
        left += 1
    elif sum_N > result:
        right -= 1
    elif sum_N < result:
        left += 1
print(cnt)